src/ol_infrastructure/infrastructure/vault/__main__.py

```diff
@@ -376,6 +423,24 @@ def cloud_init_user_data(
                 "path": "/etc/default/caddy",
                 "content": (f"DOMAIN={vault_dns_name}\n"),
             },
+            {
+                "path": "/etc/default/raft_backup",
+                "content": textwrap.dedent(
+                    f"""\
+                    export BUCKET_NAME={vault_backup_bucket}
+                    export RAFT_BACKUP_USERNAME="{vault_creds['raft_backup_username']}"
+                    export RAFT_BACKUP_PASSWORD="{vault_creds['raft_backup_password']}"
```

Comment:
Rather than passing these values through userdata and making them accessible to anything on the box, can we maybe use a local Vault agent to manage these credentials? Maybe by using EC2 auth so that there's no need to manage credentials in the script and just rely on the local agent for creating the token?

Reply:
I guess my only concern here is a boot-strapping/ chicken+egg problem? And I'm not sure if I have fully thought out what that problem might actually look like. I will investigate this because it is definitely cleaner than having a user+pass on the box, despite 0700 perms.